features/lorawan/lorastack/phy/LoRaPHY.cpp

```diff
@@ -886,30 +894,74 @@ void LoRaPHY::compute_rx_win_params(int8_t datarate, uint8_t min_rx_symbols,
                                              ((uint32_t *)phy_params.bandwidths.table)[rx_conf_params->datarate]);
     }
 
-    if (rx_conf_params->rx_slot == RX_SLOT_WIN_1) {
-        rx_conf_params->frequency = phy_params.channels.channel_list[rx_conf_params->channel].frequency;
+    switch (rx_conf_params->rx_slot) {
+        case RX_SLOT_WIN_1:
+            rx_conf_params->frequency = phy_params.channels.channel_list[rx_conf_params->channel].frequency;
+            break;
+        case RX_SLOT_WIN_BEACON:
+            max_preamble_len = BEACON_PREAMBLE_LENGTH;
+            break;
+        case RX_SLOT_WIN_UNICAST_PING_SLOT:
+        case RX_SLOT_WIN_MULTICAST_PING_SLOT:
+        case RX_SLOT_WIN_2:
+        case RX_SLOT_WIN_CLASS_C:
+            break;
+        default:
+            MBED_ASSERT(false);
```

Comment:
shouldnt these (also similar on line 964) use `MBED_ERROR` ? to report what error and msg as well ?

Reply:
After reading this again, I  see I went a different direction by adding a return status that requires the caller to  report an error. I'm unfamiliar with MBED_ERROR, so I'll look into that.